app/src/main/java/org/xbmc/kore/provider/MediaDatabase.java

```diff
@@ -88,7 +88,14 @@
                 ALBUM_GENRES + "." + MediaContract.AlbumGenres.HOST_ID + "=" + AUDIO_GENRES + "." + MediaContract.AudioGenres.HOST_ID +
                 " AND " +
                 ALBUM_GENRES + "." + MediaContract.AlbumGenres.GENREID + "=" + AUDIO_GENRES + "." + MediaContract.AudioGenres.GENREID;
-	}
+
+        /**
+         * Join to get Songs for an Artist
+         */
+        String SONGS_FOR_ARTIST_JOIN =
+                SONGS + " JOIN " + ALBUM_ARTISTS + " ON " +
+                SONGS + "." + MediaContract.Songs.ALBUMID + "=" + ALBUM_ARTISTS + "." + MediaContract.AlbumArtists.ALBUMID;
```

Comment:
I thought it was redundant as host id was added to the where clause in the builder. But after taking a closer look I noticed it isn't the same. Sorry about that.